std/experimental/allocator/building_blocks/bitmapped_block.d

```diff
@@ -229,571 +173,1462 @@ struct BitmappedBlock(size_t theBlockSize, uint theAlignment = platformAlignment
+            }
+            return result.ptr ? result.ptr[0 .. s] : null;
+        }
+
+        @trusted void[] allocateFresh(const size_t s)
+        {
+            static if (isShared)
+            {
+                lock.lock();
+                scope(exit) lock.unlock();
+            }
+
+            const blocks = s.divideRoundUp(blockSize);
+
+            void[] result = blocksFor(cast(size_t) (_freshBit / 64),
+                cast(uint) (_freshBit % 64), blocks);
+            if (result)
+            {
+                (cast(BitVector) _control)[_freshBit .. _freshBit + blocks] = 1;
+                static if (isShared)
+                {
+                    ulong localFreshBit = _freshBit;
+                    localFreshBit += blocks;
+                    _freshBit = localFreshBit;
+                }
+                else
+                {
+                    _freshBit += blocks;
+                }
+            }
+            return result;
+        }
+
+        void[] alignedAllocate(size_t n, uint a)
+        {
+            static if (isShared)
+            {
+                lock.lock();
+                scope(exit) lock.unlock();
+            }
+
+            return alignedAllocateImpl(n, a);
+        }
+
+        // If shared, this is protected by a lock inside 'alignedAllocate'
+        private void[] alignedAllocateImpl(size_t n, uint a)
+        {
+            import std.math : isPowerOf2;
+            assert(a.isPowerOf2);
+            if (a <= alignment) return allocate(n);
+
+            // Overallocate to make sure we can get an aligned block
+            auto b = allocateImpl((n + a - alignment).roundUpToMultipleOf(blockSize));
+            if (!b.ptr) return null;
+            auto result = b.roundStartToMultipleOf(a);
+            assert(result.length >= n);
+            result = result.ptr[0 .. n]; // final result
+
+            // Free any blocks that might be slack at the beginning
+            auto slackHeadingBlocks = (result.ptr - b.ptr) / blockSize;
+            if (slackHeadingBlocks)
+            {
+                deallocateImpl(b[0 .. slackHeadingBlocks * blockSize]);
+            }
+
+            // Free any blocks that might be slack at the end
+            auto slackTrailingBlocks = ((b.ptr + b.length)
+                - (result.ptr + result.length)) / blockSize;
+            if (slackTrailingBlocks)
+            {
+                deallocateImpl(b[$ - slackTrailingBlocks * blockSize .. $]);
+            }
+
+            return result;
+        }
+
+        /*
+        Tries to allocate "blocks" blocks at the exact position indicated by the
+        position wordIdx/msbIdx (msbIdx counts from MSB, i.e. MSB has index 0). If
+        it succeeds, fills "result" with the result and returns tuple(size_t.max,
+        0). Otherwise, returns a tuple with the next position to search.
+        */
+        private Tuple!(size_t, uint) allocateAt(size_t wordIdx, uint msbIdx,
```

Comment:
Document the parameters and return type in ddoc style please.
(I know it's private, but the format is well-recognized for Phobos developers and helps to grasp what's going on)

Reply:
This is also preexisting, so deferring. This PR is mostly refactoring and moving of code in spite of appearances.